diff --git a/test/test_puma_server.rb b/test/test_puma_server.rb
index e537acc0..4097a435 100644
--- a/test/test_puma_server.rb
+++ b/test/test_puma_server.rb
@@ -727,7 +727,7 @@ EOF

     sock.gets

-    assert_operator request_body_wait, :>=, 1000
+    assert_operator request_body_wait, :>=, 900
   end

   def test_request_body_wait_chunked
@@ -743,6 +743,6 @@ EOF

     sock.gets

-    assert_operator request_body_wait, :>=, 1000
+    assert_operator request_body_wait, :>=, 900
   end
 end
