diff --git a/cacheable.gemspec b/cacheable.gemspec
index 1acbed3..6a776da 100644
--- a/cacheable.gemspec
+++ b/cacheable.gemspec
@@ -25,6 +25,7 @@ Gem::Specification.new do |s|
   s.add_development_dependency "rails", ">= 4.2"
   s.add_development_dependency "activesupport"
   s.add_development_dependency "actionpack", ">= 4.1"
+  s.add_development_dependency "tzinfo-data", ">= 1.2019.3"
 
   s.add_runtime_dependency "useragent"
   s.add_runtime_dependency "msgpack"
